addons/project/models/project_task.py

```diff
@@ -225,6 +228,7 @@ def _read_group_personal_stage_type_ids(self, stages, domain, order):
     depend_on_ids = fields.Many2many('project.task', relation="task_dependencies_rel", column1="task_id",
                                      column2="depends_on_id", string="Blocked By", tracking=True, copy=False,
                                      domain="[('project_id', '!=', False), ('id', '!=', id)]")
+    depend_on_count = fields.Integer(string="Depending on Tasks", compute='_compute_depend_on_count')
```

Comment:
You can do
```suggestion
    depend_on_count = fields.Integer(string="Depending on Tasks", compute='_compute_depend_on_count', compute_sudo=True)
```
Instead of adding a sudo in the compute